libc/stdlib/stdlib_strto.c

```diff
@@ -0,0 +1,1121 @@
+		errno = 0;
+		end = NULL;
+		TEST_ASSERT_EQUAL_FLOAT(expected[i], strtof(str[i], &end));
+		TEST_ASSERT_EQUAL_PTR(str[i] + strlen(str[i]), end);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtof_min_max)
+{
+	const char floatMinString[] = "0.000000000000000000000000000000000000011754943508222875";
+	const char floatMaxString[] = "340282346638528859811704183484516925440.000000";
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MIN, strtof("1.17549435e-38", NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MIN, strtof(floatMinString, NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MAX, strtof("3.40282347e+38", NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MAX, strtof(floatMaxString, NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+/* Disabled because of #703 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/703 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#703 issue");
+#endif
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MIN, strtof("0X1.0P-126", NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+	errno = 0;
+	TEST_ASSERT_EQUAL_FLOAT(FLT_MAX, strtof("0X1.FFFFFEP+127", NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+}
+
+
+TEST(stdlib_strto, strtold_basic)
+{
+	char *end;
+	const char *str[] = { "0", "-0", "-1", "1", "+1", ".1", "3.1415", "-3.1415", "0.1234567891234567",
+		"1e0", "1e+0", "1e-0", "1e1", "-.75e+8", "-.75E+8", "3.14E+3", "3.14E-2", "3.14e+3", "-3.14e-2" };
+	const long double expected[] = { 0, 0, -1, 1, 1, 0.1, 3.1415, -3.1415, 0.1234567891234567, 1.0, 1.0, 1.0, 10.0,
+		-.75e+8, -.75E+8, 3140.0, 0.0314, 3140, -0.0314 };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		end = NULL;
+		TEST_ASSERT_EQUAL_DOUBLE(expected[i], strtold(str[i], &end));
+		TEST_ASSERT_EQUAL_PTR(str[i] + strlen(str[i]), end);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtold_basic_hex)
+{
+/* Disabled because of #703 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/703 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#703 issue");
+#endif
+	char *end;
+	const char *str[] = { "0X0.0P+0", "0X1.0P+0", "-0X1.0P+0", "0X1.0P-126", "0X1.FFFFFEP+127", "-0X1.0P-126", "-0X1.FFFFFEP+127", "0x1p-1022", "0x1.fffffffffffffp+1023" };
+	const long double expected[] = { 0.0, 1, -1, FLT_MIN, FLT_MAX, -FLT_MIN, -FLT_MAX, DBL_MIN, DBL_MAX };
+
+	for (int i = 0; i < (sizeof(str) / sizeof(str[0])); i++) {
+		errno = 0;
+		end = NULL;
+		TEST_ASSERT_EQUAL_DOUBLE(expected[i], strtold(str[i], &end));
+		TEST_ASSERT_EQUAL_PTR(str[i] + strlen(str[i]), end);
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(stdlib_strto, strtold_min_max)
+{
+	errno = 0;
+	TEST_ASSERT_EQUAL_DOUBLE(hugeDouble, strtold(hugeDoubleString, NULL));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+
+/* LDBL_MIN_MAX not declared on Phoeix_RTOS */
+#ifdef __phoenix__
```

Comment:
Are you sure? I think it could be visible as other defines from float.h:
![Screenshot from 2023-06-27 16-24-54](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/9a4bc538-153f-4647-8be7-c3c5e5563d8b)

Reply:
This way the rest of code in this test case is disabled for Phoenix so the next ifdefs have no effect.